presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -106,8 +106,21 @@ default boolean schemaExists(ConnectorSession session, String schemaName)
 
     /**
      * List table names, possibly filtered by schema. An empty list is returned if none match.
+     * Please use {@link ConnectorMetadata#listTables(ConnectorSession, Optional)} instead.
      */
-    List<SchemaTableName> listTables(ConnectorSession session, String schemaNameOrNull);
+    @Deprecated
+    default List<SchemaTableName> listTables(ConnectorSession session, String schemaNameOrNull)
+    {
+        throw new PrestoException(NOT_SUPPORTED, "This connector does not support list tables");
```

Comment:
Let's make this return an empty list. It's legitimate for a connector not to enumerate tables (we couldn't do this before since default methods are new in Java 8)